meson.build

```diff
@@ -376,9 +376,18 @@ endif
 if get_option('test')
     subdir('test')
 endif
+if get_option('compare') and default_library != 'shared'
```

Comment:
just like every other option in the autotools and meson config setups, this should error out if it was explicitly requested but is unavailable instead of silently omitting it